### DISCUSSION TITLE
How to connect using supabase as source

### DISCUSSION THREAD

**Discussion by hikrr** [nocodb/nocodb#8251] [Q&A] [answered]
![image](https://github.com/nocodb/nocodb/assets/80665253/50a26f2b-da19-4a50-8936-e47464599426)

This is my db setting page in supabse.

after I replace this string with my real password and paste it to nocodb cloud "Use Connection URL"
postgres://postgres.deqtlaxsbvexxinrjmgh:[YOUR-PASSWORD]@aws-0-ap-southeast-1.pooler.supabase.com:5432/postgres

after I press "Test Database Connection" button in nocodb, this is the error I got:
self-signed certificate in certificate chain

![image](https://github.com/nocodb/nocodb/assets/80665253/6d1abca7-8e27-4525-8b3b-348b0662d999)

I searched online and couldn't find something relevant.
I think supabase is popular and this discussion might be helpful for others as well.

Thanks for your help!

**Comment by o1lab** [comment]
Try using advanced parameters and set SSL to allowed. Between we have loads of our users connecting to supabase. Not sure what is the problem here.

<img width="577" alt="image" src="https://github.com/nocodb/nocodb/assets/5435402/52f475fa-2ee6-4e05-975b-d7224dc616fa">

**Reply by hikrr** [reply]
@o1lab

thanks for your quick reply!

![image](https://github.com/nocodb/nocodb/assets/80665253/0744e877-018a-42c9-b9bd-4fab57aa98f5)

i still have this issue.

**Reply by hikrr** [reply, marked as answer]
@o1lab and future discoverers!

I am able to connect to the db now by modifying

![image](https://github.com/nocodb/nocodb/assets/80665253/9a4f99d9-ee71-413f-ba17-73ba2f6071e3)

This is the original text:

```
{
  "title": "decent_narwhal",
  "dataSource": {
    "client": "pg",
    "connection": {
      "user": "postgres.deqtlaxsbvexxinrjmgh",
      "password": "THIS WAS MY PASSWORD",
      "database": "postgres",
      "host": "aws-0-ap-southeast-1.pooler.supabase.com",
      "port": "5432",
      "ssl": "true"
    }
  },
  "inflection": {
    "inflectionColumn": "none",
    "inflectionTable": "none"
  },
  "sslUse": "Allowed",
  "extraParameters": []
}
```

This is the new modified text:

```
{
  "title": "decent_narwhal",
  "dataSource": {
    "client": "pg",
    "connection": {
      "user": "postgres.deqtlaxsbvexxinrjmgh",
      "password": "THIS WAS MY PASSWORD",
      "database": "postgres",
      "host": "aws-0-ap-southeast-1.pooler.supabase.com",
      "port": "5432",
      "ssl": {
        "rejectUnauthorized": false
      }
    }
  },
  "inflection": {
    "inflectionColumn": "none",
    "inflectionTable": "none"
  },
  "sslUse": "Allowed",
  "extraParameters": []
}
```

and everything works now

**Reply by hikrr** [reply]
the idea is from https://github.com/nocodb/nocodb/issues/324
but i am not sure whether this will have security issues haha
really new to those db settings and security things.

**Reply by Schmedu** [reply]
I would rather download the Certificate from supabase database settings and change the config to this:
```
"ssl" : {
    "ca": "<YOUR CERTIFICATE DATA>"
}
```
